src/query/builder.js

```diff
@@ -57,9 +57,16 @@ assign(Builder.prototype, {
     return cloned;
   },
 
-  timeout(ms) {
+  timeout(ms, {cancel} = {}) {
     if(isNumber(ms) && ms > 0) {
       this._timeout = ms;
+      if (cancel) {
+        if (this.client.canCancelQuery) {
+          this._cancelOnTimeout = true;
+        } else {
+          throw new Error("Query cancelling not supported for this dialect");
```

Comment:
Is this really needed here given the default client function? I would rather have the second argument ignored for other dialects and make a note of dialect support in documentation than duplicating code.